devicetypes/smartthings/inovelli-dimmer.src/inovelli-dimmer.groovy

```diff
@@ -0,0 +1,599 @@
+						description: "Option enabled: ${it.activeDescription}\n" +
+							"Option disabled: ${it.inactiveDescription}"
+					)
+					input(
+						name: it.key,
+						type: "bool",
+						title: "Enable",
+						defaultValue: it.defaultValue == it.activeOption,
+						required: false
+					)
+					break
+				case "enum":
+					input(
+						name: it.key,
+						title: "Select",
+						type: "enum",
+						options: it.values,
+						defaultValue: it.defaultValue,
+						required: false
+					)
+					break
+				case "range":
+					input(
+						name: it.key,
+						type: "number",
+						title: "Set value (range ${it.range})",
+						defaultValue: it.defaultValue,
+						range: it.range,
+						required: false
+					)
+					break
+			}
+		}
+		// Preferences template end
+	}
+}
+
+def installed() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+
+	// Preferences template begin
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		if (it.type == "boolRange" && getPreferenceValue(it) == it.disableValue) {
+			state.currentPreferencesState."$it.key".status = "disablePending"
+		} else {
+			state.currentPreferencesState."$it.key".status = "synced"
+		}
+	}
+	readConfigurationFromTheDevice();
+// Preferences template end
+	createChildDevice("smartthings", "Child Color Selection", "${device.deviceNetworkId}:" + 3, "LEDColorConfiguration", "LEDColorConfiguration")
+}
+
+def updated() {
+	// Device-Watch simply pings if no device events received for 32min(checkInterval)
+	sendEvent(name: "checkInterval", value: 2 * 15 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+	// Preferences template begin
+	parameterMap.each {
+		if (isPreferenceChanged(it)) {
+			log.debug "Preference ${it.key} has been updated from value: ${state.currentPreferencesState."$it.key".value} to ${settings."$it.key"}"
+			state.currentPreferencesState."$it.key".status = "syncPending"
+			if (it.type == "boolRange") {
+				def preferenceName = it.key + "Boolean"
+
+				if (isNotNull(settings."$preferenceName")) {
+					if (!settings."$preferenceName") {
+						state.currentPreferencesState."$it.key".status = "disablePending"
+					} else if (state.currentPreferencesState."$it.key".status == "disabled") {
+						state.currentPreferencesState."$it.key".status = "syncPending"
+					}
+				} else {
+					state.currentPreferencesState."$it.key".status = "syncPending"
+				}
+			}
+		} else if (state.currentPreferencesState."$it.key".value == null) {
+			log.warn "Preference ${it.key} no. ${it.parameterNumber} has no value. Please check preference declaration for errors."
+		}
+	}
+	syncConfiguration()
```

Comment:
I left some comments on this PR https://github.com/SmartThingsCommunity/SmartThingsPublic/pull/37228#discussion_r455460049 about recommendations for the template code and ways that it could be improved a little. Those comments/recommendations also apply here.

Reply:
Done